aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -60,22 +60,21 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
         view: &impl StateView,
         txn: &PreprocessedTransaction,
         txn_idx: TxnIndex,
-        materialize_deltas: bool,
+        aggregator_enabled: bool,
     ) -> ExecutionStatus<AptosTransactionOutput, VMStatus> {
         let log_context = AdapterLogSchema::new(self.base_view.id(), txn_idx as usize);
-
-        match self
-            .vm
-            .execute_single_transaction(txn, &view.as_move_resolver(), &log_context)
-        {
-            Ok((vm_status, mut vm_output, sender)) => {
-                if materialize_deltas {
-                    // TODO: Integrate delta application failure.
-                    vm_output = vm_output
-                        .try_materialize(view)
-                        .expect("Delta materialization failed");
+        match self.vm.execute_single_transaction(
+            txn,
+            &view.as_move_resolver(),
+            &log_context,
+            aggregator_enabled,
+        ) {
+            Ok((vm_status, vm_output, sender)) => {
+                if let PreprocessedTransaction::UserTransaction(_) = txn {
```

Comment:
awesome for fixing this flow!

I think you can do here: matches!(txn, PreprocessedTransaction::UserTransaction(_)) or the appropriate matches syntax. Useful because you can && that with aggregator_enabled to avoid nested ifs.

Could also put the whole predicate inside assert now, or have if matches && !aggregator_enabled followed by the current assert - I don't have a strong preference. But let's add a 1-line comment to the assert, smt like "with aggregators disabled, there should be no delta outputs"

Reply:
Done.